### PR TITLE
Support additional request data types in `dio` package

### LINKED ISSUE #2982: I can't see any request parameters or additional information
### Platform

Flutter Mobile Android

### Obfuscation

Enabled

### Debug Info

Enabled

### Doctor

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.29.2, on macOS 14.4.1 23E224 darwin-arm64, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[!] Xcode - develop for iOS and macOS (Xcode 15.3)
    ! CocoaPods 1.15.2 out of date (1.16.2 is recommended).
        CocoaPods is a package manager for iOS or macOS platform code.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/to/platform-plugins
      To update CocoaPods, see https://guides.cocoapods.org/using/getting-started.html#updating-cocoapods
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.3.2)
[✓] Android Studio (version 2023.2)
[✓] VS Code (version 1.93.0)
[✓] Connected device (4 available)
    ! Error: Browsing on the local area network for iPhone特色. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
[!] Network resources
    ✗ A network error occurred while checking "https://github.com/": Connection refused

! Doctor found issues in 2 categories.

### Version

8.14.2

### Steps to Reproduce

```dart
final dio = Dio();
  dio.addSentry();

  final transaction = Sentry.startTransaction(
    'dio-web-request',
    'request',
    bindToScope: true,
  );

  try {
    final response = await dio
        .get<Map<String, Object?>>('https://www.google.com/idontexist');

    print(response.toString());

    transaction.status =
        SpanStatus.fromHttpStatusCode(response.statusCode ?? -1);
  } catch (exception) {
    transaction.throwable = exception;
    transaction.setData("request_data_error", jsonEncode(body));
    transaction.status = const SpanStatus.internalError();
  } finally {
    await transaction.finish();
  }
```

### Expected Result

I hope that when an exception occurs, I can see the request parameters or my customized "request_data_error" information

### Actual Result

Image

### Are you willing to submit a PR?

None

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

- `JSON`
- `List<int>`
- `num`, `bool`
- `FormData`
- `Multipart`

## :bulb: Motivation and Context

Closes #2982

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: packages/dio/lib/src/dio_event_processor.dart [hunk 1]
```diff
@@ -49,27 +51,100 @@ class DioEventProcessor implements EventProcessor {
...
       }
+    } else if (data is Uint8List) {
+      // Handle Uint8List (typed byte array)
+      if (_options.maxRequestBodySize.shouldAddBody(data.length)) {
+        return data;
+      }
+    } else if (data is num || data is bool) {
+      if (_options.maxRequestBodySize != MaxRequestBodySize.never) {
+        return data;
+      }
```
[the remark this answers is not in the document]
Comment (buenaflor) [reply packages/dio/lib/src/dio_event_processor.dart:84]: valid concern, afaict we can remove the Uint8List check @denrase

**Comment by denrase** [comment]
https://github.blog/changelog/2025-07-11-upcoming-changes-to-macos-hosted-runners-macos-latest-migration-and-xcode-support-policy-updates/#macos-latest-migration-begins-august-4-2025

Ok, we cannot tell at this point if `macos-latest` is `macos-14` or `macos-15`, from which only the latter has iOS 18.0. So I will set this explicitly to `macos-15`, which should resolve the issue.

**Comment by denrase** [comment]
Ok, seems to be an issue with the runner after all: 

https://github.com/actions/runner-images/issues/12758

File: packages/dio/lib/src/dio_event_processor.dart [hunk 2]
```diff
@@ -49,27 +50,95 @@ class DioEventProcessor implements EventProcessor {
...
+    } else if (data is! String &&
+        Transformer.isJsonMimeType(requestOptions.contentType)) {
+      try {
+        final jsonSize = jsonEncode(data).codeUnits.length;
+        if (_options.maxRequestBodySize.shouldAddBody(jsonSize)) {
+          return data;
+        }
+      } catch (e) {
+        return null;
+      }
```
Comment (buenaflor) [inline packages/dio/lib/src/dio_event_processor.dart]: I think we ran into this problem before, I guess the issue is that we run jsonEncode(data) without length validations, e.g data could be extremely large and we try to process it in jsonEncode.

Could you check if we do something like this in other parts of the code

Comment (denrase) [reply packages/dio/lib/src/dio_event_processor.dart]: Could not find anything similar to your description. Also looked through dio code, and I don't see how we could get the correct size without `jsonEncode`. Do you have a suggestion how to work around this?

Comment (denrase) [reply packages/dio/lib/src/dio_event_processor.dart]: Added your suggestions with the sink + counting bytes. Also handling edge cases where we can bail out early.

**Review by buenaflor — COMMENTED** [review]
I think tests are missing where if `sendDefaultPii` is false it doesnt attach headers

**Review by buenaflor — APPROVED** [review]
